Source/Plugins/Core/com.equella.core/js/tsrc/mainui/routes.tsx

```diff
@@ -29,24 +29,29 @@ import LoginNoticeConfigPage from "../loginnotice/LoginNoticeConfigPage";
 import FacetedSearchSettingsPage from "../settings/Search/facetedsearch/FacetedSearchSettingsPage";
 import SearchPage from "../search/SearchPage";
 
-export interface OEQRouteComponentProps<T = any>
-  extends RouteComponentProps<T> {
+export interface OEQRouteComponentProps<T> extends RouteComponentProps<T> {
   updateTemplate(edit: TemplateUpdate): void;
   redirect(to: LocationDescriptor): void;
   setPreventNavigation(b: boolean): void;
   refreshUser(): void;
 }
 
-export interface OEQRoute {
+type To = (uuid: string, version: number) => string;
+
+export interface OEQRoute<T> {
   component?:
-    | React.ComponentType<OEQRouteComponentProps<any>>
-    | React.ComponentType<any>;
-  render?: (props: OEQRouteComponentProps<any>) => React.ReactNode;
+    | React.ComponentType<OEQRouteComponentProps<T>>
+    | React.ComponentType<T>;
+  render?: (props: OEQRouteComponentProps<T>) => React.ReactNode;
   path?: string;
   exact?: boolean;
   sensitive?: boolean;
   strict?: boolean;
-  to?: any;
+  to?: string | To;
+}
+
+export interface Routes {
+  [route: string]: OEQRoute<TemplateUpdateProps>;
 }
 
 export const routes = {
```

Comment:
So with the new `Routes` type, should we now make this a bit more explicit with:

```suggestion
export const routes: Routes = {
```